user-guide/src/overview.md

```diff
@@ -0,0 +1,24 @@
+# Overview
+
+In short, when writing a daemon or a service, we have the *"muscle"* of the
+application ‒ whatever we write the daemon for. And we have a whole lot of
+infrastructure around that: logging, command line parsing, configuration. And
+while there are Rust libraries for all that, one needs nontrivial amount of
+boilerplate code to bridge all this together.
+
+Spirit aims to be this bridge. 
+
+It takes care of things like
+
+- signal handling and the application lifecycle
+- combining multiple pieces of configuration together with command line
+  overrides
+- it allows for reloading the configuration at runtime
+- application metrics
+
+In short, it is the glue nobody really wants to write every time. It doesn’t
+do much itself, it just ties readily available libraries together.
+
+The aim is to save time with the boring stuff and to provide the boring stuff
+with some bells and whistles one wouldn’t really care to write were it just
+for this one daemon.
```

Comment:
The wording here feels a bit complex. After reading it twice, it makes sense, but it would be nice if this could be simplified.

Also, would it fit somewhere noting that, while there's a recommended way to build the daemon, one of spirit's goal is to be flexible enough to deviate from that way as needed by the application, not force the user into using it the prescribed way or not using it at all?